### ISSUE TITLE
Undefined references to java.io.ObjectInputStream when linking

### ISSUE THREAD

**Issue by espectalll** [gluonhq/client-maven-plugin#101]
Trying to compile a project on Debian Sid, I got the following error:

```
[INFO] --- client-maven-plugin:0.1.10:link (default-cli) @ app ---
[Sat Jan 11 00:40:39 CET 2020][INFO] [SUB] -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lgthread-2.0 -pthread -lglib-2.0 -lXtst
started linking
done linking
Linking failed. Details from linking below:
Command was: gcc /home/xerz/Projects/app/target/client/x86_64-linux/gvm/app/launcher.o /home/xerz/Projects/app/target/client/x86_64-linux/gvm/tmp/SVM-1578699520414/app.app.o -ljava -lnio -lzip -lnet -ljvm -lstrictmath -lz -ldl -lj2pkcs11 -lsunec -lextnet -lstdc++ -rdynamic -llibchelper -lpthread -Wl,--whole-archive -lprism_es2 -lglass -lglassgtk3 -ljavafx_font -ljavafx_font_freetype -ljavafx_font_pango -ljavafx_iio -Wl,--no-whole-archive -lGL -lX11 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lfreetype -lpangoft2-1.0 -lgthread-2.0 -lstdc++ -lz -lXtst -o /home/xerz/Projects/app/target/client/x86_64-linux/app -L/home/xerz/.opt/graalvm/lib/svm/clibraries/linux-amd64 -L/home/xerz/.gluon/substrate/javaStaticSdk/14-ea+5/linux-x86_64/labs-staticjdk/lib/static -L/home/xerz/.gluon/substrate/javafxStaticSdk/14-ea+gvm4/linux-x86_64/sdk/lib
/usr/bin/ld: /home/xerz/Projects/app/target/client/x86_64-linux/gvm/tmp/SVM-1578699520414/app.app.o:(.data+0x1a8): undefined reference to `Java_java_io_ObjectInputStream_bytesToFloats'
/usr/bin/ld: /home/xerz/Projects/app/target/client/x86_64-linux/gvm/tmp/SVM-1578699520414/app.app.o:(.data+0xa90): undefined reference to `Java_java_io_ObjectInputStream_bytesToDoubles'
collect2: error: ld returned 1 exit status
java.lang.RuntimeException: Linking failed
	at com.gluonhq.NativeLinkMojo.execute(NativeLinkMojo.java:55)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
```

If this is due to some unsupported Java feature in `native-image`, help for replacing the failing parts will be welcome

**Comment by espectalll** [comment]
Related: https://github.com/oracle/graal/issues/460

**Comment by espectalll** [comment]
It's funny, because I cannot find any object or method related to Serializable or ObjectInputStream in my source code. Since I don't know of any debugger that can set a breakpoint on an abstract class, I've went ahead and created a new project where I created sample code that could trigger anything, but I cannot find any clues. The `launcher.c` does fine in the sample project but not in the original one and I don't know where else to look at.

**Comment by espectalll** [comment]
OK, this one is getting on my nerves. I removed basically all of the source code, refactored the whole thing, deleted every single dependency, and stripped it down to the most barebone project possible. I manually deleted `target/`, `.gluon` and even `.m2`. It still doesn't work. [Here's a copy of the failing source code](https://github.com/gluonhq/client-maven-plugin/files/4049544/app.zip) if anyone wants to look into it.
